tasks/ca.yml

```diff
@@ -55,6 +55,39 @@
   args:
     creates: "{{ ca_ca_dir }}/ca.key"
 
+- name: Check if CA certificate exists
+  ansible.builtin.stat:
+    path: "{{ ca_ca_dir }}/ca.crt"
+  register: ca_certificate
+  when: ca_renew | bool
+
+- name: Verify CA certificate will not expire soon
+  community.crypto.x509_certificate_info:
+    path: "{{ ca_ca_dir }}/ca.crt"
+    valid_at:
+      check_period: "{{ ca_check_valid_time }}"
+  register: ca_expire
+  when:
+    - ca_renew | bool
```

Comment:
I like that you have this extra safety feature. Renewing a whole CA can be a huge process and you don't want that to happen by accident.